DeusEx/Classes/Player.uc

```diff
@@ -9,3 +9,144 @@ function bool AddInventory( inventory NewItem )
+        }
+    }
+
+    if ((damageType == 'Shot') || (damageType == 'Sabot') || (damageType == 'Exploded') || (damageType == 'AutoShot'))
+    {
+        // go through the actor list looking for owned BallisticArmor
+        // since they aren't in the inventory anymore after they are used
+        if (UsingChargedPickup(class'BallisticArmor'))
+        {
+            skillLevel = SkillSystem.GetSkillLevelValue(class'SkillEnviro');
+            newDamage *= 0.5 * skillLevel;
+        }
+    }
+
+    if (damageType == 'HalonGas')
+    {
+        if (bOnFire && !bCheckOnly)
+            ExtinguishFire();
+    }
+
+    if ((damageType == 'Shot') || (damageType == 'AutoShot'))
+    {
+        if (AugmentationSystem != None)
+            augLevel = AugmentationSystem.GetAugLevelValue(class'AugBallistic');
+
+        if (augLevel >= 0.0)
+            newDamage *= augLevel;
+    }
+
+    if (damageType == 'EMP')
+    {
+        if (AugmentationSystem != None)
+            augLevel = AugmentationSystem.GetAugLevelValue(class'AugEMP');
+
+        if (augLevel >= 0.0)
+            newDamage *= augLevel;
+    }
+
+    if ((damageType == 'Burned') || (damageType == 'Flamed') ||
+        (damageType == 'Exploded') || (damageType == 'Shocked'))
+    {
+        if (AugmentationSystem != None)
+            augLevel = AugmentationSystem.GetAugLevelValue(class'AugShield');
+
+        if (augLevel >= 0.0)
+            newDamage *= augLevel;
+    }
+
+    //
+    // Reduce or increase the damage based on the combat difficulty setting, do this before SetDamagePercent for the UI display
+    // because we don't want to show 100% damage reduction but then do the minimum of 1 damage
+    if ((damageType == 'Shot') || (damageType == 'AutoShot') ||
+        damageType == 'Flamed' || damageType == 'Burned')
+    {
+        newDamage *= CombatDifficulty;
+        oldDamage *= CombatDifficulty;
+
+        // always take at least one point of damage
+        if ((newDamage <= 1) && (Damage > 0))
+            newDamage = 1;
+        if ((oldDamage <= 1) && (Damage > 0))
+            oldDamage = 1;
+    }
+
+    if (newDamage < oldDamage)
+    {
+        if (!bCheckOnly)
+        {
+            //make sure to factor the rounding into the percentage
+            pct = 1.0 - ( Float(Int(newDamage)) / Float(Int(oldDamage)) );
+            SetDamagePercent(pct);
+            ClientFlash(0.01, vect(0, 0, 50));
+        }
+        bReduced = True;
+    }
+    else
+    {
+        if (!bCheckOnly)
+            SetDamagePercent(0.0);
+    }
+
+    adjustedDamage = Int(newDamage);
+
+    return bReduced;
+}
+
+function CatchFire( Pawn burner )
+{
+    local bool doSetTimer;
+    if (bOnFire==false && Region.Zone.bWaterZone==false)
+        doSetTimer = true;
+
+    _CatchFire(burner);
+
+    // set the burn timer, tick the burn every 4 seconds instead of 1 so that the player can actually survive it
+    if(doSetTimer)
+        SetTimer(4.0, True);
```

Comment:
one quarter the tick rate for the burning damage, but since I added scaling the burning damage with difficulty setting, Normal difficulty is 1.25 times, so the player ends up taking 6 damage per tick instead of 5 now

this allows the player to survive burning on Normal difficulty without using medkits or anything as long as you had 100 torso health